cibuildwheel/windows.py

```diff
@@ -154,6 +154,16 @@ def setup_setuptools_cross_compile(
     # set the platform name
     map_plat = {"32": "win32", "64": "win-amd64", "ARM64": "win-arm64"}
     plat_name = map_plat[python_configuration.arch]
+
+    # Set environment variable so that setuptools._distutils.get_platform()
+    # identifies the target, not the host
+    vscmd_arg_tgt_arch = {"32": "x86", "64": "x64", "ARM64": "arm64"}
+    current_tgt_arch = vscmd_arg_tgt_arch[python_configuration.arch]
+    if env.get("VSCMD_ARG_TGT_ARCH", current_tgt_arch) != current_tgt_arch:
+        msg = f"VSCMD_ARG_TGT_ARCH must be set to {current_tgt_arch}, got {env['VSCMD_ARG_TGT_ARCH']}. Make sure you setup MSVC targeting the right architecture."
```

Comment:
Given that preconfiguring MSVC is now incompatible with our default setting of building both 32 and 64 bit windows in one cibuildwheel session, I feel we should suggest that users don't preconfigure MSVC using vcvarsall/msvc-dev-cmd.

```suggestion
        msg = f"VSCMD_ARG_TGT_ARCH must be set to {current_tgt_arch!r}, got {env['VSCMD_ARG_TGT_ARCH']!r}. If you're setting up MSVC yourself (e.g. using vcvarsall.bat or msvc-dev-cmd), make sure to target the right architecture. Alternatively, run cibuildwheel without configuring MSVC, and let the build backend handle it."
```

Reply:
Thanks everyone, that's great feedback. I'll put it all together and finalise the PR on Thu/Fri - right now I'm snowed under until then with work & personal stuff. Hope that timeframe is OK for you...

Reply:
Some build backends don't like it not being set up, I think. Usually people add this because something is broken. However, on an error message, I think this is useful.

Reply:
No rush, and I've been adding and accepting changes to see if I can get it working.